### ISSUE TITLE
If \if, \else or \endif is in a list, the subsequent list entry is not formatted as list entry but in "raw format"

### ISSUE THREAD

**Issue by Christian390** [doxygen/doxygen#8817]
**Describe the bug**
I have lists with conditional list entries. The conditional entries are inside an \if ... \endif or \if ... \else ... \endif construct. With Doxygen 1.8.18 (on Windows 10 PC) everything was as expected: The list entries were shown or not, depending on the condition.
With Doxygen 1.9.2 this still is true but with the difference that the list entry on the next line after \if, \else or \endif is not formatted as list entry but in "raw format".

I seems to be the problem that \if, \else, \endif no longer expand to something with a new line character at the end. I think that because I found two workarounds by placing a newline after these commands. See the aliases in the attached Doxyfile 'DoxyfileWithWorkaround'.

**Expected behavior**
List entries in the next line after \if, \else and \endif are formatted as list entry (with a bullet point or number) and not in raw format ("- Text" or "-# Text").

**Screenshots**
What I get is this:
![image](https://user-images.githubusercontent.com/54361383/136186230-fd80dd5d-a969-42b1-b285-2996f415c6b9.png)
What I want is this:
![image](https://user-images.githubusercontent.com/54361383/136186382-b9c8e007-50a0-45d2-b39f-5a5ef9f19840.png)

**To Reproduce**
[DoxyIfElse.zip](https://github.com/doxygen/doxygen/files/7292944/DoxyIfElse.zip)

Use DoxyfileWithoutWorkaround first to see the full problem.
After that use DoxyfileWithWorkaround to see how my workarounds (the aliases) work around the problem.
After that please fix the bug so that I do not need these workarounds :-) Thanks a lot!

**Version**
doxygen -v
1.9.2 (caa4e3de211fbbef2c3adf58a6bd4c86d0eb7cb8)
Platform : Windows 10, 64 bit

**Additional context**
Here the input that produced the screenshots (its in the reproduce-zip too):
```
/**

\mainpage

ASDF is an enabled section<br>
QWER is \b no enabled section

Without workaround

-# '1'
-# '2'
\if ASDF
-# '3' IF-CASE ASDF
\else
-# '3' ELSE-CASE ASDF
\endif
-# '4'
\if QWER
-# '5' IF-CASE QWER
\else
-# '5' ELSE-CASE QWER
\endif
-# '6'
-# '7'

With workaround \\myif

-# '1'
-# '2'
\myif{ASDF}
-# '3' IF-CASE ASDF
\else
-# '3' ELSE-CASE ASDF
\endif
-# '4'
\myif{QWER}
-# '5' IF-CASE QWER
\else
-# '5' ELSE-CASE QWER
\endif
-# '6'
-# '7'

With workaround \\if ... \\fi

-# '1'
-# '2'
\if ASDF \fi
-# '3' IF-CASE ASDF
\else
-# '3' ELSE-CASE ASDF
\endif
-# '4'
\myif{QWER}
-# '5' IF-CASE QWER
\else
-# '5' ELSE-CASE QWER
\endif
-# '6'
-# '7'

*/
```

**Comment by albert-github** [comment]
I've just pushed a proposed patch, pull request #8819

**Comment by albert-github** [comment]
Code has been integrated in master on GitHub (please don't close the issue as this will be done at the moment of an official release).

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.9.3.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
